File: leetcode/ReverseWords.java
/**
 * https://leetcode.com/problems/reverse-words-in-a-string/
 */
class ReverseWords {
    public static String reverseWords(String s) {

        if (s.isEmpty()) {
            return "";
        }

        String[] sArray = s.trim().split("\\s+");

        StringBuilder sb = new StringBuilder();


        String sep = "";
        for (int i = sArray.length - 1; i >= 0; i--) {
            sb.append(sep + sArray[i]);
            sep = " ";
        }

        return sb.toString();
    }

    public static void main(String[] args) {
        System.out.println(reverseWords("some randomw string"));
    }
}

File: leetcode/ReverseWords.scala
/*
 ** https://leetcode.com/problems/reverse-words-in-a-string/
 */
object Solution {
    def reverseWords(s: String): String = {

        if(s.isEmpty) {
            return ""
        }

        val sArray = s.trim.split("\\s+")
        var output = "";
        var sep = "";

        val arrayRange = sArray.size - 1

        for (i <- arrayRange to 0 by -1) {
            output = output + sep + sArray(i)
            sep = " "
        }

        return output
    }
}
